android/src/main/java/com/dexterous/flutterlocalnotifications/FlutterLocalNotificationsPlugin.java

```diff
@@ -348,6 +349,30 @@ private static IconCompat getIconFromSource(Context context, String iconPath, Ic
         return icon;
     }
 
+    private static void setVisibility(NotificationDetails notificationDetails, NotificationCompat.Builder builder) {
+        if (notificationDetails.visibility == null) {
+            return;
+        }
+
+        int visibility;
+        switch (notificationDetails.visibility) {
+            case 0: // Private
+                visibility = NotificationCompat.VISIBILITY_PRIVATE;
+                break;
+            case 1: // Public
+                visibility = NotificationCompat.VISIBILITY_PUBLIC;
+                break;
+            case 2: // Secret
+                visibility = NotificationCompat.VISIBILITY_SECRET;
+                break;
+
+            default:
+                System.out.println("FlutterLocalNotificationsPlugin.setVisibility - case not handled: " + notificationDetails.visibility);
```

Comment:
I understand the reasoning behind adding this but would like see this removed. Logging to the console can confuse users of the plugin (has happened in the past). I would rather see us go with an IllegalArgumentException. I've seen a thread on the main Flutter repository on best practices for plugins in these scenarios but can't find it right now. At the least, there's an example in one of the official plugins [here](https://github.com/flutter/plugins/blob/017a57e1d3f8e3da7ac40ee35bbaaca059bc6cb6/packages/path_provider/android/src/main/java/io/flutter/plugins/pathprovider/StorageDirectoryMapper.java#L48)